index.js

```diff
@@ -30,6 +52,16 @@ class TrezorKeyring extends EventEmitter {
     this.paths = {};
     this.deserialize(opts);
     TrezorConnect.manifest(TREZOR_CONNECT_MANIFEST);
+    TrezorConnect.getFeatures().then((features) => {
+      console.log('features', features);
+      if (features.success) {
+        this.model = features.payload.model;
+      }
+    });
+  }
+
+  getModel() {
+    return this.model;
```

Comment:
:+1: It's pretty smart to cache the `async` call result! For some reason, it never occurred to me when I was trying to get it working myself.

It might be better to use something like `getEIP1559Support() {}` instead of `getModel() {}`.

That way, when the Trezor Model One adds EIP-1559 support, you can change the function to always `return true;`, and remove the `TrezorConnect.getFeatures()` call.

```suggestion
  getEIP1559Support() {
    // TODO, replace with `return true;`
    // when Trezor Model One has added support (expected December 2021)
    if (!this.model) {
      throw Error("Cannot find Trezor model, please ensure it's connected");
    }
    return this.model === "T";
```

Reply:
Nice suggestion. Thanks

Reply:
Actually, I went back to `getModel`. This will allow us to get make a change in `metamask-extension` when the time comes, and also gives us more flexibility. In the future, maybe other features will depend on the model, and the more general method can accomodate that

Reply:
Makes sense! I know EIP-712 support should be coming soonish, and it will only (initially) be on the Model T, so having a `getModel()` would make it easier.